From: Raphael Kubo da Costa <rakuco@FreeBSD.org>
Date: Sun, 22 Jan 2012 19:03:05 +0000
Subject: cli7zip: Completely remove RootNodeSwitch.
X-Git-Url: http://quickgit.kde.org/?p=ark.git&amp;a=commitdiff&amp;h=0e67d6db9e7a5a0f90fcacaee11042b89b79d96b
---
cli7zip: Completely remove RootNodeSwitch.

Follow-up to 45c6c6f5bcff6bf754fccd6377836e025a6baade, which removed the
definition of RootNodeSwitch but did not remove the reference in
ExtractArgs, resulting in a crash whenever a 7z archive was extracted.
---


--- a/plugins/cli7zplugin/cliplugin.cpp
+++ b/plugins/cli7zplugin/cliplugin.cpp
@@ -53,7 +53,7 @@ ParameterList CliPlugin::parameterList()
         p[ListProgram] = p[ExtractProgram] = p[DeleteProgram] = p[AddProgram] = QLatin1String( "7z" );
 
         p[ListArgs] = QStringList() << QLatin1String( "l" ) << QLatin1String( "-slt" ) << QLatin1String( "$Archive" );
-        p[ExtractArgs] = QStringList() << QLatin1String( "$PreservePathSwitch" ) << QLatin1String( "$PasswordSwitch" ) << QLatin1String( "$RootNodeSwitch" ) << QLatin1String( "$Archive" ) << QLatin1String( "$Files" );
+        p[ExtractArgs] = QStringList() << QLatin1String( "$PreservePathSwitch" ) << QLatin1String( "$PasswordSwitch" ) << QLatin1String( "$Archive" ) << QLatin1String( "$Files" );
         p[PreservePathSwitch] = QStringList() << QLatin1String( "x" ) << QLatin1String( "e" );
         p[PasswordSwitch] = QStringList() << QLatin1String( "-p$Password" );
         p[FileExistsExpression] = QLatin1String( "already exists. Overwrite with" );
